pkg/3scale/amp/operator/deployment_config_reconciler.go

```diff
@@ -127,3 +128,46 @@ func NewCreateOnlyDCReconciler() *CreateOnlyDCReconciler {
 func (r *CreateOnlyDCReconciler) IsUpdateNeeded(desired, existing *appsv1.DeploymentConfig) bool {
 	return false
 }
+
+func DeploymentConfigReconcileDeploymentTriggerImageChangeParamsName(desired, existing *appsv1.DeploymentConfig, logger logr.Logger) bool {
+	update := false
+	findDeploymentTriggerOnImageChange := func(triggerPolicies []appsv1.DeploymentTriggerPolicy) (int, error) {
+		result := -1
+		for i := range triggerPolicies {
+			if triggerPolicies[i].Type == appsv1.DeploymentTriggerOnImageChange {
+				if result != -1 {
+					return -1, fmt.Errorf("found more than one imageChangeParams Deployment trigger policy")
+				}
+				result = i
+			}
+		}
+
+		if result == -1 {
+			return -1, fmt.Errorf("no imageChangeParams deployment trigger policy found")
+		}
+
+		return result, nil
+	}
+
+	desiredDeploymentTriggerImageChangePos, err := findDeploymentTriggerOnImageChange(desired.Spec.Triggers)
+	if err != nil {
+		panic(fmt.Sprintf("unexpected: '%s' in DeploymentConfig '%s'", err, desired.Name))
```

Comment:
In case the ImageStream imagechange trigger has not been found or more than one imagecange trigger has been found we panic and finish the execution

Reply:
already commented